### ISSUE TITLE
Timezone logged incorrectly when between -1 and -59 minutes offset from UTC

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3515]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3515|
|Type|defect|
|Created|2008-10-29 22:57:42Z|
|Branch|https://github.com/twisted/twisted/tree/negative-tzoffset-3515|

As reported initially on [#3513](https://github.com/twisted/twisted/issues/3513), if a tzoffset is negative and less than one hour, it is incorrectly written to the log as the absolute value of the offset.

Attachments:

* [log.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/b0a/b0ac3e4ef3dcc9c2a966d4b8af44f6340ce449f6/a8bf44c1276b2ac10ec0acf104bfd8996b0a5763.diff) (2742 bytes) - added by wangchun on 2008-10-30 01:49:12Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3515 3515
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_negative_tzoffset_3515 branches/negative-tzoffset-3515
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1225321062000000 1225321062000000
changetime__1225645761000000 1225645761000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25189]) Branching to 'negative-tzoffset-3515'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [25190]) fix the tzoffset reporting

refs [#3515](https://github.com/twisted/twisted/issues/3515)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="wangchun's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|wangchun commented|
|-|-|

Patch updated and attached.

**Comment by twisted-trac** [comment]
|<img alt="wangchun's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|wangchun commented|
|-|-|

According to RFC 2822:

  The form "+0000" SHOULD be used to indicate a time zone at Universal Time.  Though "-0000" also indicates Universal Time, it is used to indicate that the time was generated on a system that may be in a local time zone other than Universal Time and therefore indicates that the date-time contains no information about the local time zone.

I believe the formatTime got ```localtime``` here, and we should use ```%Y-%m-%dT%H:%M:%S-0000``` here for London.

But in RFC 3339:

  If the time in UTC is known, but the offset to local time is unknown, this can be represented with an offset of "-00:00".  This differs semantically from an offset of "Z" or "+00:00", which imply that UTC is the preferred reference point for the specified time.  RFC2822 [IMAIL-UPDATE] describes a similar convention for email.

It seems we should not use "-0000" for a known time zone.

UNIX strftime returns +0000 in London, so the patch attached.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

The proposed fix is in the branch indicated in the ticket description area - `branches/negative-tzoffset-3515`.  I didn't base it off of your original patch because it was easier to start over than to remove the unrelated changes.  If you want to suggest further changes, the best way to do so is with a patch against the branch, rather than against trunk.  Please don't include changes not related to this ticket (you can file more tickets for other changes :), eg the addition of 'T' to the middle of the time format.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @exarkun|
|-|-|

Great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [25286]) Merge negative-tzoffset-3515

Author: exarkun
Reviewer: therve
Fixes: [#3515](https://github.com/twisted/twisted/issues/3515)

Fix the formatting done by `FileLogObserver` of timezone offsets
between -1 and -59 minutes (ie, log events emitted in Liberia
before May 1, 1972 or in Ireland between 1880 and October 1, 1916).

Previously these timezone offsets would be incorrectly reported as
positive rather than negative.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
